### DISCUSSION TITLE
Ingress to HTTPProxy does not work

### DISCUSSION THREAD

**Discussion by VA-PLE** [projectcontour/contour#4297] [Q&A] [answered]
Hi. I needed HTTPProxy enhancements and I started translating my Ingress rules into HTTPProxy
 

    kind: Ingress
    apiVersion: networking.k8s.io/v1
    metadata:
      name: test-contour-test
      namespace: bm-eu-prod
      annotations:
        ingress.kubernetes.io/force-ssl-redirect: 'false'
        kubernetes.io/ingress.class: contour
    spec:
      tls:
        - hosts:
            - adx-eu.test.io
          secretName: test.io
      rules:
        - host: adx-eu.test.io
          http:
            paths:
              - path: /
                pathType: Prefix
                backend:
                  service:
                    name: auction-gateway
                    port:
                      number: 9000

 
converted to

 

    apiVersion: projectcontour.io/v1
    kind: HTTPProxy
    metadata:
      name: test-contour-test-new
      namespace: bm-eu-prod
    spec:
      virtualhost:
        fqdn: adx-eu.test.io
        tls:
          secretName: test.io
      routes:
        - conditions:
          - prefix: /
          services:
            - name: auction-gateway
              port: 9000

 

when I do curl -i adx-eu.test.io/auction/status (Ingress)

    HTTP/1.1 200 OK
    date: Wed, 26 Jan 2022 14:02:37 GMT
    content-length: 0
    x-envoy-upstream-service-time: 1
    server: envoy

when I do curl -i adx-eu.test.io/auction/status (HTTPProxy)

    HTTP/1.1 404 Not Found
    vary: Accept-Encoding
    date: Wed, 26 Jan 2022 13:45:33 GMT
    server: envoy
    content-length: 0

kubectl describe HTTPProxy/test-contour-test-new -n bm-eu-prod

    Name:         test-contour-test-new
    Namespace:    bm-eu-prod
    Labels:       <none>
    Annotations:  <none>
    API Version:  projectcontour.io/v1
    Kind:         HTTPProxy
    Metadata:
      Creation Timestamp:  2022-01-26T14:08:28Z
      Generation:          1
      Managed Fields:
        API Version:  projectcontour.io/v1
        Fields Type:  FieldsV1
        fieldsV1:
          f:spec:
            .:
            f:routes:
            f:virtualhost:
              .:
              f:fqdn:
              f:tls:
                .:
                f:secretName:
        Manager:         dashboard
        Operation:       Update
        Time:            2022-01-26T14:08:28Z
      Resource Version:  1718581833
      UID:               6915e995-a2bf-4a8c-be17-82bc8b707be2
    Spec:
      Routes:
        Conditions:
          Prefix:  /
        Services:
          Name:  auction-gateway
          Port:  9000
      Virtualhost:
        Fqdn:  adx-eu.test.io
        Tls:
          Secret Name:  test.io
    Events:             <none>

what am i doing wrong? k8s 1.21.1;contour:1.18.2 from helm chart https://github.com/bitnami/charts/tree/master/bitnami/contour

**Comment by sunjayBhatia** [comment, marked as answer]
It looks like your conversion is valid, however the bitnami helm chart configures Contour by default with an Ingress class to ensure Contour only applies configuration for Ingress resources that apply to it. Your Ingress resource above has an Ingress class annotation and you will want to add that same class via annotation or the `ingressClassName` spec field: https://projectcontour.io/docs/v1.18.2/config/api/#projectcontour.io/v1.HTTPProxySpec
